### ISSUE TITLE
Definition with wildcards generates anomaly "grounding a non evar-free term"

### ISSUE THREAD

**Issue by anton-trunov** [coq/coq#7265]
#### Version

Coq master (3e7863e9369d38537685576a8642dbe0c062d0c5)

#### Operating system

macOS 10.13.3 High Sierra

#### Description of the problem

Please consider the following minimized example:

```coq
Definition foo : _ := _.
```

It generates the following anomaly:

>Error: Anomaly "grounding a non evar-free term"
>Please report at http://coq.inria.fr/bugs/.

It's a regression since Coq 8.7.2, which correctly reports:

>Error: Cannot infer this placeholder of type "Type".

**Comment by ejgallego** [comment]
This is due to #6454 , and due to code using `EConstr.to_constr` incorrectly without properly flagging it. Will fix. [Not an 8.8 regression tho]

**Comment by maximedenes** [comment]
@ejgallego any update on this?

**Comment by ejgallego** [comment]
Didn't have time to look into this, anyways running `-debug` on should point the location and the problem should be easy to fix.

**Comment by anton-trunov** [comment]
current Coq master
```
$ bin/coqtop -debug
Welcome to Coq 192.168.1.136:/Users/anton/prj/coq/coq,master (fd146ca38202c9843b4240cbdac0ae75f57e4d67)

Coq < Definition foo : _ := _.
Toplevel input, characters 0-24:
> Definition foo : _ := _.
> ^^^^^^^^^^^^^^^^^^^^^^^^
Error: Anomaly "in econstr: grounding a non evar-free term"
Please report at http://coq.inria.fr/bugs/.
frame @ file "toplevel/vernac.ml", line 172, characters 2-60
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "stm/stm.ml", line 600, characters 33-68
raise @ file "pervasives.ml", line 32, characters 17-33
frame @ file "stm/stm.ml", line 2629, characters 4-60
frame @ file "stm/stm.ml", line 2519, characters 4-105
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "stm/stm.ml", line 935, characters 6-10
frame @ file "stm/stm.ml", line 2392, characters 16-43
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "stm/stm.ml", line 1088, characters 12-91
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "vernac/vernacentries.ml", line 2332, characters 4-36
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "lib/flags.ml", line 17, characters 14-17
raise @ unknown
frame @ file "vernac/vernacentries.ml", line 2299, characters 8-666
frame @ file "lib/flags.ml", line 99, characters 20-44
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "lib/flags.ml", line 17, characters 14-17
frame @ file "vernac/comDefinition.ml", line 112, characters 4-62
frame @ file "vernac/comDefinition.ml", line 93, characters 14-97
frame @ file "clib/option.ml", line 100, characters 19-24
frame @ file "vernac/comDefinition.ml", line 93, characters 60-86
raise @ file "lib/loc.ml", line 89, characters 16-23
```

vs

Coq 8.8.0:
```
$ coqtop -debug
Welcome to Coq 8.8.0 (April 2018)

Coq < Definition foo : _ := _.
Toplevel input, characters 17-18:
> Definition foo : _ := _.
>                  ^
Error: Cannot infer this placeholder of type
"Type".
frame @ file "toplevel/vernac.ml", line 172, characters 2-66
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "stm/stm.ml", line 600, characters 33-68
raise @ file "pervasives.ml", line 32, characters 17-33
frame @ file "stm/stm.ml", line 2635, characters 4-55
frame @ file "stm/stm.ml", line 2525, characters 4-100
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "stm/stm.ml", line 934, characters 6-10
frame @ file "stm/stm.ml", line 2398, characters 16-43
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "stm/stm.ml", line 1087, characters 12-91
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "vernac/vernacentries.ml", line 2343, characters 4-36
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "lib/flags.ml", line 17, characters 14-17
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "vernac/vernacentries.ml", line 2310, characters 8-666
frame @ file "lib/flags.ml", line 100, characters 20-44
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "lib/flags.ml", line 17, characters 14-17
frame @ file "vernac/comDefinition.ml", line 131, characters 18-38
frame @ file "pretyping/pretyping.ml", line 381, characters 2-49
frame @ file "set.ml", line 339, characters 35-38
raise @ file "clib/exninfo.ml", line 65, characters 2-15
```

**Comment by anton-trunov** [comment]
Fixed by #7449
